src/core/infer_request.h

```diff
@@ -304,9 +304,13 @@ class InferenceRequest {
 
   // Add an override input to the request. If 'input' is non-null
   // return a pointer to the newly added input.
+  // FIXME passing batch size is special handling for backend API.
```

Comment:
So once all backends move to new API we can get rid of the batch_size parameter?

Reply:
Yes, the shape will change to contain batch size if support batching, since all backend can only access input properties via backend API which only provides shape with batch dim